### ISSUE TITLE
[subgraph_rewriter] Add match_filter for the matches

### ISSUE THREAD

**Issue by jerryzh168** [pytorch/pytorch#69577]
## 🚀 Feature
Sometimes we may want to further filtering for the matches. like https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/fx/quantized_fusion_patterns_and_replacements.py#L23

we can add a match_filters argument to the replace_pattern function:

```
def replace_pattern(
        gm: GraphModule,
        pattern: Callable,
        replacement: Callable,
        is_match_filters: Optional[List[Callable]] = None) -> List[Match]:
   ...
```

feel free to take a look at https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/fx/subgraph_rewriter_FORKED_DO_NOT_USE.py for a possible implementation

**Comment by SherlockNoMad** [comment]
Yes, this can be done by adding a Callable filter to SubgraphMatcher. 
Will to in latter iteration.
